lib/ansible/modules/packaging/language/pip.py

```diff
@@ -222,19 +222,34 @@
 import re
 import sys
 import tempfile
+from distutils.versionpredicate import VersionPredicate
```

Comment:
So I've tried a specifier, supported by pip and it turns out that distutils' parser don't support just a major version

```python-traceback
In [56]: vp = VersionPredicate('cherrypy(==4)')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-56-b8d12ec3c098> in <module>()
----> 1 vp = VersionPredicate('cherrypy(==4)')

~/.pyenv/versions/3.6.5/lib/python3.6/distutils/versionpredicate.py in __init__(self, versionPredicateStr)
    114                 raise ValueError("expected parenthesized list: %r" % paren)
    115             str = match.groups()[0]
--> 116             self.pred = [splitUp(aPred) for aPred in str.split(",")]
    117             if not self.pred:
    118                 raise ValueError("empty parenthesized list in %r"

~/.pyenv/versions/3.6.5/lib/python3.6/distutils/versionpredicate.py in <listcomp>(.0)
    114                 raise ValueError("expected parenthesized list: %r" % paren)
    115             str = match.groups()[0]
--> 116             self.pred = [splitUp(aPred) for aPred in str.split(",")]
    117             if not self.pred:
    118                 raise ValueError("empty parenthesized list in %r"

~/.pyenv/versions/3.6.5/lib/python3.6/distutils/versionpredicate.py in splitUp(pred)
     24         raise ValueError("bad package restriction syntax: %r" % pred)
     25     comp, verStr = res.groups()
---> 26     return (comp, distutils.version.StrictVersion(verStr))
     27
     28 compmap = {"<": operator.lt, "<=": operator.le, "==": operator.eq,

~/.pyenv/versions/3.6.5/lib/python3.6/distutils/version.py in __init__(self, vstring)
     38     def __init__ (self, vstring=None):
     39         if vstring:
---> 40             self.parse(vstring)
     41
     42     def __repr__ (self):

~/.pyenv/versions/3.6.5/lib/python3.6/distutils/version.py in parse(self, vstring)
    135         match = self.version_re.match(vstring)
    136         if not match:
--> 137             raise ValueError("invalid version number '%s'" % vstring)
    138
    139         (major, minor, patch, prerelease, prerelease_num) = \

ValueError: invalid version number '4'

```

The one from `pkg_resources`, which is being used by pip, works though:
```python
from pkg_resources import Requirement
rq = Requirement('cherrypy=4.0')

In [58]: rq.specifier.contains('4')
Out[58]: True

In [59]: rq.specifier.contains('5')
Out[59]: False

In [60]: rq.specifier.contains('4.0.0')
Out[60]: True

In [61]: rq.specifier.contains('4.0.1')
Out[61]: False

In [62]: Requirement.parse('cherrypy>=4,<6').specifier.contains('5')
Out[62]: True

In [63]: Requirement.parse('cherrypy>=4,<6').specifier.contains('6')
Out[63]: False

```

@HD650 please switch to `pkg_resources.Requirement`

Reply:
You are right, VersionPredicate only supports full version number but users may input some simplified version number.